lib/improver/nbhood/circular_kernel.py

```diff
@@ -0,0 +1,361 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing neighbourhood processing utilities."""
+
+import iris
+from iris.exceptions import CoordinateNotFoundError
+import numpy as np
+import scipy.ndimage.filters
+
+from improver.percentile import PercentileConverter
+from improver.utilities.spatial import (
+    convert_distance_into_number_of_grid_cells)
+
+# Maximum radius of the neighbourhood width in grid cells.
+MAX_RADIUS_IN_GRID_CELLS = 500
+
+
+class Utilities(object):
+    """
+    Contains methods useful to multiple circular_kernel plugins.
+    """
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<Utilities>')
+        return result
+
+    @staticmethod
+    def circular_kernel(fullranges, ranges, weighted_mode):
```

Comment:
Usually, but it can be any two dimensions. I think having both would make it easier to add temporal neighbourhoods if we ever wanted to. Perhaps this is sufficiently unlikely to ignore...